test/onnx/test_fx_to_onnx_with_onnxruntime.py

```diff
@@ -891,15 +906,13 @@ def _test_fake_tensor_mode_exporter(
         ) as tmp_checkpoint_file:
             # Dump state_dict to a file to simulate how HuggingFace model is initialized.
             # The file will be loaded via .load_state_dict(...)
-            state_dict = real_model.state_dict()
+            state_dict = _extract_all_buffers_and_params(real_model)
             torch.save(state_dict, tmp_checkpoint_file.name)
 
             with torch.onnx.enable_fake_mode() as fake_context:
                 fake_args = create_args()
                 fake_kwargs = create_kwargs()
                 fake_model = create_model()
-                if load_checkpoint_during_init:
```

Comment:
I think the purpose of this flag was to cover both use case scenarios:
- Load model by `from_pretrained`.
- Load model by `load_state_dict` from checkpoint file.

Make sure both are still covered after your change.

Reply:
It does. `load_state_dict ` is only needed in FxSymbolicTracer. I guess dynamo somehow manages to have the info without `load_state_dict`.

Reply:
That's not the case I think, both are emulating how a pretrained model is loaded in pytorch. It is unrelated to dynamo. `from_pretrained` is a transformers api, while `load_state_dict` is the standard [pytorch api](https://pytorch.org/tutorials/beginner/saving_loading_models.html#saving-loading-model-for-inference).